### PR TITLE
ONT-897 add ontID format check in several functions

### PR DESCRIPTION
**Pull request by kunxian-xia**
- initContractAdmin
- transfer
- assignOntIDsToRole
- delegate

### CODE REVIEW

File: smartcontract/service/native/auth/auth.go [hunk 1]
```diff
@@ -70,6 +71,9 @@ func InitContractAdmin(native *native.NativeService) ([]byte, error) {
 	}
 	invokeAddr := cxt.ContractAddress
 
+	if !ontid.VerifyID(param.AdminOntID) {
+		return utils.BYTE_FALSE, nil
```
Comment (Honglei-Cong) [inline smartcontract/service/native/auth/auth.go]: do we need provide some error information?

Comment (kunxian-xia) [reply smartcontract/service/native/auth/auth.go]: push error out?

Comment (Honglei-Cong) [reply smartcontract/service/native/auth/auth.go]: just a question :)

return BYTE_FALSE, any error-info?

Comment (kunxian-xia) [reply smartcontract/service/native/auth/auth.go]: In this case, the param provided is not valid, so I choose to return false for there is no internal-error happened. Maybe I should add some error-info log?

File: smartcontract/service/native/ontid/utils.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func decodeID(data []byte) ([]byte, error) {
 	return data[1:], nil
 }
 
-func verifyID(id []byte) bool {
+func VerifyID(id []byte) bool {
```
Comment (laizy) [inline smartcontract/service/native/ontid/utils.go]: remove this duplicated code and use the one in `account/identity.go`

Comment (kunxian-xia) [reply smartcontract/service/native/ontid/utils.go]: already done : )

**Review by laizy — APPROVED** [review]

**Pull request merged**